### PR TITLE
Can register relations under a custom name, defaults to base_name

### PR DESCRIPTION
**Pull request by mcls**
Took a stab at #137 

I'm not very familiar with your test suite, so if this needs to be tested in other places, please tell me :smile:

### CODE REVIEW

File: spec/integration/repositories/extending_relations_spec.rb [hunk 1]
```diff
@@ -47,4 +47,19 @@ def freaking_cool?
...
+          register_as :users
+        }
+        Class.new(ROM::Relation[:memory]) {
+          base_name :guests
+          register_as :admins
+        }
+      end
+    end
+  end
 end
```
Comment (solnic) [inline spec/integration/repositories/extending_relations_spec.rb]: This should live in `spec/unit/rom/setup_spec.rb`

Comment (mcls) [inline spec/integration/repositories/extending_relations_spec.rb]: Ok, I'll update that

File: lib/rom/relation.rb [hunk 2]
```diff
@@ -42,6 +42,10 @@ def initialize(dataset, registry = {})
       @__registry__ = registry
     end
 
+    def self.register_as(value = ROM::ClassMacros::Undefined)
+      super || self.base_name
+    end
+
```
Comment (solnic) [inline lib/rom/relation.rb]: Yep this is what we want. This reminds me - could you also move `Undefined` to `ROM::Undefined`? We should use it in many places :)

Comment (mcls) [inline lib/rom/relation.rb]: :+1:

**Comment by mcls** [comment]
I just noticed that the `RelationAlreadyDefinedError` only gets raised when using the setup DSL.
Should I move that to `ROM::Setup::Finalize`?

https://github.com/rom-rb/rom/blob/85226fae8624e278703175aa1c23ceabff2e9c74/lib/rom/setup_dsl/setup.rb#L21

**Comment by solnic** [comment]
Yep good idea to move it :+1:

File: lib/rom/support/undefined.rb [hunk 3]
```diff
@@ -0,0 +1,3 @@
+module ROM
+  Undefined = Object.new.freeze
+end
```
Comment (solnic) [inline lib/rom/support/undefined.rb]: ah, one small thing - could you move that to `rom.rb`? doesn't really deserve to be in a separate file (and requires are slow :))

**Pull request merged**